source/common/tracing/BUILD

```diff
@@ -12,12 +12,16 @@ envoy_cc_library(
     name = "http_tracer_lib",
     srcs = [
         "http_tracer_impl.cc",
-        "lightstep_tracer_impl.cc",
-    ],
+    ] + select({
+        "//bazel:disable_lightstep_tracing": [],
+        "//conditions:default": ["lightstep_tracer_impl.cc"],
+    }),
     hdrs = [
         "http_tracer_impl.h",
-        "lightstep_tracer_impl.h",
-    ],
+    ] + select({
+        "//bazel:disable_lightstep_tracing": [],
+        "//conditions:default": ["lightstep_tracer_impl.h"],
+    }),
     external_deps = ["lightstep"],
```

Comment:
"select" is actually a type that's implemented by skylark. I haven't figured out how (or if) to get at its contents from within build extensions. Obviously we'll need to figure it out long term, I just haven't done that here.